src/gluonnlp/vocab/vocab.py

```diff
@@ -203,6 +238,44 @@ def _index_counter_keys(self, counter, unknown_token, special_tokens, max_size,
                 self._idx_to_token.append(token)
                 self._token_to_idx[token] = len(self._idx_to_token) - 1
 
+    def _expose_tokens_as_attributes(self, identifiers_to_tokens):
+        for identifier, token in identifiers_to_tokens.items():
+            if token not in self:
+                raise ValueError(
+                    '\'{}\' is not part of the vocabulary. '
+                    '\'{}\' cannot identify a non-existing token.'.format(
+                        token, identifier))
+            if identifier.startswith('_'):
+                raise ValueError('It is not allowed to use identifiers starting with '
+                                 'underscore. In Python identifier names beginning with '
+                                 'underscore are internal.')
+            if hasattr(self, identifier):
+                raise ValueError('vocab.{} already exists. '
+                                 'Please choose a different identifier for token {}'
+                                 .format(identifier, token))
+            setattr(self, identifier, token)
+
+    def _sort_index_according_to_user_specification(self, token_to_idx):
+        # Sanity checks
+        if not set(token_to_idx.keys()).issubset(self.token_to_idx.keys()):
+            raise ValueError('User-specified token_to_idx mapping can only contain '
+                             'tokens that will be part of the vocabulary.')
+        if len(set(token_to_idx.values())) != len(token_to_idx):
+            raise ValueError('User-specified indices must not contain duplicates.')
+        if min(token_to_idx.values()) < 0 or max(token_to_idx.values()) >= len(self.token_to_idx):
+            raise ValueError('User-specified indices must not be < 0 or >= the number of tokens '
+                             'that will be in the vocabulary.')
```

Comment:
Could you also include the value of `len(self.token_to_idx)` in the err msg? That helps the user to correct his idx value